src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using SpikeCore.Data.Models;
+using SpikeCore.Web.Models;
+using SpikeCore.Web.TokenProviders;
+
+namespace SpikeCore.Web.Controllers
+{
+    /// <summary>
+    /// Provides a controller for authenticating users via validating tokens issued by the 
+    /// <see cref="PasswordlessLoginTokenProvider{TUser}"/>.
+    ///
+    /// This is based on https://www.scottbrady91.com/ASPNET-Identity/Implementing-Mediums-Passwordless-Authentication-using-ASPNET-Core-Identity.
+    /// </summary>
+    public class AuthenticationController : Controller
+    {
+        private readonly UserManager<SpikeCoreUser> _userManager;
+
+        public AuthenticationController(UserManager<SpikeCoreUser> userManager)
+        {
+            _userManager = userManager;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> TokenCallback(string email, string token)
+        {
+            var user = await _userManager.FindByEmailAsync(email);
+
+            var isValid =
+                await _userManager.VerifyUserTokenAsync(user, "PasswordlessLoginProvider", "passwordless-auth", token);
```

Comment:
Would be nice to either have const's or `nameof(PasswordlessLoginTokenProvider)`, although interestingly enough, nameof on a generic type needs an type argument that it then proceeds to ignore, e.g. `nameof(Meh<int>)` -> `"Meh"`

Reply:
After talking about this on IRC, I'll go ahead and drop the generics on the type... we really only have one type of user anyway.